### PR TITLE
[WebNN EP] Support axes and fix some validation for Resize

### PR DESCRIPTION
**Pull request by Honry**
- Supports arbitrary axes for Resize opset 18+
- Check all inputs and attributes more carefully

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc [hunk 1]
```diff
@@ -105,55 +208,46 @@ void ResizeOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const N
...
-
   std::vector<float> scales;
-  std::vector<int32_t> sizes;
-  std::vector<float> scales_hw;
-  std::vector<int32_t> sizes_hw;
-  std::vector<int32_t> axes;
-  std::string scales_name = GetTensorName(input_defs, 2);
+  std::vector<int64_t> sizes;
+  std::vector<uint32_t> webnn_sizes;
+  std::vector<int64_t> axes = GetAxes(helper, 4);  // We alreday checked input shape is 4d in IsOpSupportedImpl.
```
[the remark this answers is not in the document]
Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc:230]: Also, the comment claims the input shape is already checked, but I'm having trouble confirming that. 🤔

**Review by fdwr — COMMENTED** [review]
Thanks Wanming.

File: onnxruntime/core/providers/webnn/builders/helper.h [hunk 2]
```diff
@@ -147,6 +156,16 @@ inline bool ReadScalarTensorData(const onnx::TensorProto& tensor, emscripten::va
   return true;
 }
 
+inline bool IsEmptyTensor(const InitializedTensorSet& initializers, const std::string name) {
```
[the remark this answers is not in the document]
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/helper.h]: :( No matching member function for call to `at`.

File: onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc [hunk 3]
```diff
@@ -58,20 +75,65 @@ bool GetResizeScales(const InitializedTensorSet& initializers,
...
   const auto& sizes_tensor = *initializers.at(input_defs[3]->Name());
-  if (sizes_tensor.dims_size() != 1 || sizes_tensor.dims()[0] != 4)
+  // Number of elements of sizes tensor.
+  const auto num_of_sizes = sizes_tensor.dims()[0];
+  if (sizes_tensor.dims_size() != 1) {
+    LOGS(logger, ERROR) << "'sizes' should be a 1D tensor.";
     return false;
+  }
+
+  if (has_axes && num_of_sizes != 2) {
```
[the remark this answers is not in the document]
Comment (Honry) [reply onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc:126]: Good proposal, this is actually a requirement for supporting `Resize` for input shape less than 4D, currently implementation only supports 4D input.

Besides, in onnxruntime, layout transformation will transpose `sizes` and `scales` as well if it is NHWC input layout, (I fixed a minor bug to restrict it to only transpose 4D `sizes` and `scales`, https://github.com/microsoft/onnxruntime/pull/21954/files, PTAL :) ), this increases the complexity a bit, so can we move this requirement to a follow-up?

**Review by Honry — COMMENTED** [review]
@fdwr, thanks for your comments, I've addressed them % the support for input shape < 4D. PTAL, thanks!

File: onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (fdwr) [reply onnxruntime/core/providers/webnn/builders/impl/resize_op_builder.cc]: Sorry, thought I clicked the Expand all button, but must not have (as my eyes and Ctrl+F were not coming across it).

**Review by fdwr — DISMISSED** [review]
👍

**Review by fdwr — APPROVED** [review]
👍

**Review by fdwr — APPROVED** [review]
👍 Reapproving.

**Pull request merged**